drivers/storage/portworx/component/autopilot.go

```diff
@@ -968,8 +970,12 @@ func generateAPSaToken(clusterNamespace string, expirationSeconds int64) (*authv
 	return tokenResp, nil
 }
 
-func isTokenRefreshRequired(secret *v1.Secret) (bool, error) {
-	if len(secret.Data) == 0 || len(secret.Data[v1.ServiceAccountTokenKey]) == 0 {
+// isSecretUpdateRequired returns true if secret is not present
+// isSecretUpdateRequired returns true if token expiry is not set
+// isSecretUpdateRequired returns true if token is expired
+// isSecretUpdateRequired returns false if token is not expired
```

Comment:
nit: avoid explaining the whole logic in the comment since it will be difficult to keep it up-to-date as the function changes. We should just explain the background that is not obvious from reading the code.